src/Http/Headers/src/MediaTypeHeaderValue.cs

```diff
@@ -645,7 +646,7 @@ private static int GetMediaTypeExpressionLength(StringSegment input, int startIn
             }
             else
             {
-                mediaType = input.Substring(startIndex, typeLength) + ForwardSlashCharacter + input.Substring(current, subtypeLength);
+                mediaType = string.Concat(input.AsSpan().Slice(startIndex, typeLength), "/", input.AsSpan().Slice(current, subtypeLength));
```

Comment:
@BrennanConroy can you propose an API `StringSegment.AsSpan(int start, int length)` on dotnet/runtime

Reply:
https://github.com/dotnet/runtime/issues/50428